lib/puppet/type/ec2_securitygroup.rb

```diff
@@ -37,6 +37,21 @@ def insync?(is)
     end
   end
 
+  newproperty(:egress, :array_matching => :all) do
+    desc 'rules for egress traffic'
+    def insync?(is)
+      for_comparison = Marshal.load(Marshal.dump(should))
+      parser = PuppetX::Puppetlabs::AwsIngressRulesParser.new(for_comparison)
```

Comment:
If the `AwsIngressRulesParser` can parse both ingress and egress rules, maybe it should be renamed too? What do you think of `AwsSecurityGroupRulesParser` ?